### PR TITLE
Upgrade Select2 to v4

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is BC break.

Part of: https://github.com/sonata-project/SonataAdminBundle/issues/7049

## Changelog

```markdown
### Changed
- Upgrade to Select2 ^4.0.
```

### CODE REVIEW

File: src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig [hunk 1]
```diff
@@ -137,64 +124,34 @@ file that was distributed with this source code.
...
+                    return {% block sonata_type_model_autocomplete_dropdown_item_format -%}
+                        {% if safe_label|default(false) %}
+                            '<div class="{{ dropdown_item_css_class }}">'+item.label+'<\/div>'
+                        {% else %}
+                            jQuery('<div class="{{ dropdown_item_css_class }}">').text(item.label).prop('outerHTML')
+                        {% endif %}
+                    {%- endblock %}; // format of one dropdown item
+                },
+                templateSelection: function (item) {
+                    // Select2 v4 BC select pre-selection.
```
Comment (jordisala1991) [inline src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:138]: I have to revisit this one.

File: src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig [hunk 2]
```diff
@@ -58,13 +52,7 @@ file that was distributed with this source code.
     <script>
```
Comment (core23) [inline src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:52]: Would be great if we can completely remove the on page javascript an move this to the `admin.js` file. That would be necesarry if you use a strict CSP.

Comment (jordisala1991) [reply src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:52]: Yes, I have added that to the issue #7049 

It needs a lot of work for some templates, we should probably take a look at previous PRs that tried it.

File: docs/reference/configuration.rst [hunk 3]
```diff
@@ -182,12 +182,12 @@ Full Configuration Options
                     - bundles/sonataadmin/dist/bootstrap/bootstrap.min.css
                     - bundles/sonataadmin/dist/components-font-awesome/css/font-awesome.min.css
                     - bundles/sonataadmin/dist/ionicons/css/ionicons.min.css
-                    - bundles/sonataadmin/dist/admin-lte/css/AdminLTE.min.css
+                    - bundles/sonataadmin/dist/admin-lte/css/AdminLTE-without-plugins.min.css
```
Comment (jordisala1991) [inline docs/reference/configuration.rst:185]: Default admin-lte has custom styles for select2, fullcalendar and bootstrap-social.

From those, we are only using select2, and the 4 version has imcompatible styles for it: https://github.com/ColorlibHQ/AdminLTE/issues/668

**Review by VincentLanglet — APPROVED** [review]

**Review by phansys — APPROVED** [review]

**Pull request merged**